src/main/java/com/google/codeu/servlets/LogoDetectServlet.java

```diff
@@ -25,6 +34,83 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
     Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
     List<BlobKey> blobKeys = blobs.get("image");
-    response.sendRedirect("/home.html");
+
+    if (blobKeys != null && !blobKeys.isEmpty()) {
+      BlobKey blobKey = blobKeys.get(0);
+      ImagesService imagesService = ImagesServiceFactory.getImagesService();
+      ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
+      String imageUrl = imagesService.getServingUrl(options);
+      byte[] blobBytes = getBlobBytes(blobstoreService, blobKey);
+      try {
+        List<String> logos = detectLogos(blobBytes);
+        String keyWord = logos.get(0); //Chose the first logo for now
+        response.sendRedirect("/info-present.html?kw=" + keyWord);
+      } catch (Exception e) {
```

Comment:
This is pretty broad and at least requires some explanation (i.e. comment). If there is a specific error `detectLogos` throws that you want to catch -- then catch that specific error. In any case you need to log the error somewhere, so that you can look at it.

Reply:
Actually, I would suggest removing the catch and let the error bubble up. If it becomes frequent and annoying, then it needs to be addressed, but at that time you would at least know what the problem is. If the exception just silently ignored, it will just appear unstable for no particular reason.